Onboarding/QA/tests/profile-generic_test.exs

```diff
@@ -0,0 +1,263 @@
+          message: "ℹ️ Our team of experts has put together loads of health information for you. To quickly get a selection of the info that is valuable to you, share more information about yourself.\n\nReady to share?",
+          buttons: [
+            %Btn.Next{title: "Yes, let's go"},
+            %Btn.Next{title: "Not right now"}
+          ]
+        }
+      ]
+    }
+
+    remind_later = %ContentPage{
+      slug: "mnch_onboarding_remind_later",
+      title: "Remind_later",
+      parent: "test",
+      wa_messages: [
+        %WAMsg{
+          message: "*All good. I’ll check in with you about this another time.* 🗓️\r\n\r\nFor now, I recommend having a look at some of the most popular topics on {MyHealth}.\r\n\r\n👇🏽 What do you want to do now?",
+          buttons: [
+            %Btn.Next{title: "See popular topics"}
+          ]
+        }
+      ]
+    }
+
+    assert :ok =
+             FakeCMS.add_pages(wh_pid, [
+               %Index{slug: "test", title: "test"},
+               error_pg,
+               progress_30_generic,
+               progress_100_generic,
+               why_personal_info_1,
+               remind_later
+             ])
+
+    # Return the adapter.
+    FakeCMS.wh_adapter(wh_pid)
+  end
+
+  defp real_or_fake_cms(step, base_url, _auth_token, :real),
+    do: WH.allow_http(step, base_url)
+
+  defp real_or_fake_cms(step, base_url, auth_token, :fake),
+    do: WH.set_adapter(step, base_url, setup_fake_cms(auth_token))
+
+  defp setup_flow() do
+    # When talking to real contentrepo, get the auth token from the API_TOKEN envvar.
+    auth_token = System.get_env("API_TOKEN", "CRauthTOKEN123")
+    kind = if auth_token == "CRauthTOKEN123", do: :fake, else: :real
+
+    flow_path("profile-generic")
+    |> FlowTester.from_json!()
+    |> real_or_fake_cms("https://content-repo-api-qa.prk-k8s.prd-p6t.org/", auth_token, kind)
+    |> FlowTester.set_global_dict("config", %{"contentrepo_token" => auth_token})
+  end
+
+  # This lets us have cleaner button/list assertions.
+  def indexed_list(var, labels) do
+    Enum.with_index(labels, fn lbl, idx -> {"@#{var}[#{idx}]", lbl} end)
+  end
+
+  # The common case for buttons.
+  defmacro button_labels(labels) do
+    quote do: unquote(indexed_list("button_labels", labels))
+  end
+
+  # The common case for lists.
+  defmacro list_items(labels) do
+    quote do: unquote(indexed_list("list_items", labels))
+  end
+
+  describe "profile generic" do
+    test "30% complete" do
+      setup_flow()
+      |> FlowTester.set_contact_properties(%{"year_of_birth" => "1988", "province" => "Western Cape", "area_type" => "", "gender" => "male"}) # Basic Information
+      |> FlowTester.set_contact_properties(%{"relationship_status" => "", "education" => "", "socio_economic" => "", "other_children" => ""}) # Personal Information
+      |> FlowTester.set_contact_properties(%{"dma_01" => "answer", "dma_02" => "", "dma_03" => "", "dma_04" => "", "dma_05" => ""}) # Daily Life
```

Comment:
Can some of these contact properties go in `setup_flow`? Or perhaps a helper function that we can call with some overrides for specific tests?

Reply:
Perhaps use these helpers in other tests as well? 😉